substrate/frame/revive/rpc/src/client.rs

```diff
@@ -226,7 +234,43 @@ impl Client {
 			fee_history_provider: FeeHistoryProvider::default(),
 			chain_id,
 			max_block_weight,
-		})
+		};
+
+		Ok(client)
+	}
+
+	/// Build the genesis block (block 0) on-demand for RPC queries.
+	///
+	/// This method reconstructs the EVM genesis block using the same logic as
+	/// `ensure_genesis_block()`, but returns it for use in RPC responses.
+	async fn build_genesis_block(&self, genesis_block: &SubstrateBlock) -> Option<Block> {
+		let substrate_hash = genesis_block.hash();
+		let runtime_api = self.runtime_api(substrate_hash);
+
+		let gas_limit = runtime_api.block_gas_limit().await.inspect_err(|err| {
+			log::error!(target: LOG_TARGET, "Failed to get block gas limit for block #{}: {err:?}", genesis_block.number());
+		}).ok()?;
```

Comment:
won't this always fail for a real chain since the block_gas_limit was not available at genesis?